torchdata/datapipes/iter/util/header.py

```diff
@@ -53,3 +56,65 @@ def __len__(self) -> int:
                 "The actual value may be smaller if the actual length of source_datapipe is smaller than the limit."
             )
             return self.limit
+
+
+@functional_datapipe("limit")
+class LimiterIterDataPipe(IterDataPipe[T_co]):
+    """
+    Limits the number of samples that the source DataPipe can yield based the input as well as the number of
+    processes and workers if distribute- or multi-processing is enabled (functional name: ``limit``).
+
+    The number of samples that will be yielded equals ``num_take // (num_processes * worker_info.num_workers)``.
+
+    Args:
+        source_datapipe: IterDataPipe with samples of data that will be yielded
+        num_take: the number of samples that you wish to yield across all processes and workers, should be less than
+            or equal to the ``len(source_datapipe)``
+
+    Example:
+        >>> source_dp = IterableWrapper(range(20))
+        >>> limiter_dp = source_dp.limit(5)
+        >>> list(limiter_dp)
+        [0, 1, 2, 3, 4]
+    """
+
+    def __init__(self, source_datapipe: IterDataPipe[T_co], num_take: int) -> None:
+        super().__init__()
+        self.source_datapipe: IterDataPipe[T_co] = source_datapipe
+        self.num_take: int = num_take
+        self.num_processes: int = 1
+        if dist.is_available() and dist.is_initialized():
+            self.num_processes = dist.get_world_size()
```

Comment:
I'm a little worried about this bit. Is it a safe assumption that a user wants to use distributed if it is initialized?

Reply:
This is a similar heuristic as what we do in `DistributedSampler` https://github.com/pytorch/pytorch/blob/89d6f3e6094f36adade286e85b5c3b7d05c4003b/torch/utils/data/distributed.py#L63-L70

As I mentioned in another comment, this can be factored out in a different element, but there are also other things to consider

Reply:
Please add TODO to pull this information from the ReadingService